extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/blocking/BlockingExecutionHandler.java

```diff
@@ -37,8 +37,7 @@ public void handle(Promise<Object> event) {
          * The lock object is assumed to be the request's listener
          */
         synchronized (lock) {
-            final Context previous = Context.current();
-            grpcContext.attach();
+            Context previous = grpcContext.attach();
```

Comment:
This could fix `Detaching context which was not attached.` error, but it brings a warning `Context in storage not the expected context, Scope.close was not called correctly` from https://github.com/quarkusio/quarkus/blob/6d24576fafb2b0fec3efb16f98a7abcff6f84fe8/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/QuarkusContextStorage.java#L40

I am not sure about the root cause, might need some help.

Reply:
oops, I might've merged prematurely. Are you sure it's this change that brings the warning you linked?

Reply:
Yeah, I am pretty sure this PR brings this warning.

Reply:
I am not sure what to do next. Should we revert this merged PR or fix this warning in the subsequent PR?

Reply:
I think this change is good.

Do you have a small project that reproduces the warning?

Reply:
You can follow the steps from #23603 with changes in this PR.

Reply:
I'm quite busy, it would help me a lot if you could put up a ready reproducer, with a test for which the warning is printed.
If you don't have time for it, it's absolutely okay :)

Reply:
OK, I could prepare a temp public repo for it.

Reply:
You can use this repo https://github.com/XSAM/quarkus-issue-23603-reproducer to reproduce this warning. It uses a local Quarkus build `999-SNAPSHOT`.

Reply:
The warning is not printed without this change only because the error that you are fixing happens right before it and this iteration of closing the context is not invoked. I think it's safe to say this change didn't break anything

Reply:
Thanks, @michalszynkiewicz. Is there a plan to fix this warning?

Reply:
Contributions are welcome ;)
Can you create an issue for it?

Reply:
> Contributions are welcome ;)

I would love to contribute the fix, but I am new in Quarkus; this may go beyond my ability.

> Can you create an issue for it?

Sure, see #23805